sql/1715118159.sql

```diff
@@ -0,0 +1,108 @@
+-- +migrate Up
+-- +migrate StatementBegin
+create or replace function convoy.take_token(_key text, _rate integer, _bucket_size integer) returns boolean
+    language plpgsql
+as
+$$
+DECLARE
+    next_min timestamptz;
+    _can_take BOOLEAN;
+    row RECORD;
+BEGIN
+    SELECT * FROM convoy.token_bucket WHERE key = _key FOR UPDATE SKIP LOCKED LIMIT 1 INTO row;
```

Comment:
@jirevwe nitpick, but for perf reasons, we should only select the columns we need. :)